code/game/alternate_appearance.dm

```diff
@@ -72,43 +88,62 @@ GLOBAL_LIST_EMPTY(active_alternate_appearances)
 	if(transfer_overlays)
 		I.copy_overlays(target)
 
-	hud_icons = list(appearance_key)
-	add_to_hud(target, I)
+	add_atom_to_hud(target)
+	target.set_hud_image_active(appearance_key, exclusive_hud = src)
+
 	if((options & AA_TARGET_SEE_APPEARANCE) && ismob(target))
-		add_hud_to(target)
+		show_to(target)
+
 	if(add_ghost_version)
 		var/image/ghost_image = image(icon = I.icon , icon_state = I.icon_state, loc = I.loc)
 		ghost_image.override = FALSE
 		ghost_image.alpha = 128
 		ghost_appearance = new /datum/atom_hud/alternate_appearance/basic/observers(key + "_observer", ghost_image, NONE)
 
+	if(ismovable(target))
+		var/atom/movable/AM = target
+		if(AM.bound_overlay)
+			mimic(AM.bound_overlay)
+
 /datum/atom_hud/alternate_appearance/basic/Destroy()
 	. = ..()
 	QDEL_NULL(image)
 	target = null
 	if(ghost_appearance)
 		QDEL_NULL(ghost_appearance)
 
-/datum/atom_hud/alternate_appearance/basic/add_to_hud(atom/A)
+/datum/atom_hud/alternate_appearance/basic/proc/mimic(atom/movable/openspace/mimic)
```

Comment:
Nit: `/atom/movable/openspace/mimic/M` or something, the /openspace type is abstract and not relevant here.

Reply:
Unintentional, was meant to be typed to mimic.